Modules/grpmodule.c

```diff
@@ -119,16 +120,51 @@ grp_getgrgid_impl(PyObject *module, PyObject *id)
         }
         Py_DECREF(py_int_id);
     }
+#ifdef HAVE_GETGRGID_R
+    Py_BEGIN_ALLOW_THREADS
+    int status, bufsize = NSS_BUFLEN_GROUP;
+    struct group *grpbuf = NULL;
+
+    p = malloc(sizeof(struct group));
+    buf = malloc(bufsize);
+
+    do {
+        status = getgrgid_r(gid, p, buf, bufsize, &grpbuf);
+        if(grpbuf == NULL && status == ERANGE) {
+            free(buf);
+            if((bufsize << 1) > (NSS_BUFLEN_GROUP << 10)) {
```

Comment:
What is this check for?

Reply:
This is to make sure we are not allocating too big of a buffer (its a maximum cap)